src/ir_passes/local_port_elim.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+
+use crate::{
+    ir::{Access, Command, Component, Connect, Ctx, Info, PortIdx},
+    ir_visitor::{Action, Visitor},
+};
+
+/// Eliminates local ports from the IR.
+#[derive(Default)]
+pub struct LocalPortElim {
```

Comment:
The existing flow also checks to make sure that all the local ports have exactly one write to them. This means they error if there are multiple writes to the same port location. We should recreate that behavior.

Also, does this assume that all local ports are exactly one element long? That assumption is not true right?

Reply:
It is true because this happens post bundle-elim

Reply:
> The existing flow also checks to make sure that all the local ports have exactly one write to them. This means they error if there are multiple writes to the same port location. We should recreate that behavior.

Doesn't bind check or whatever in the first place already check that all ports are only assigned to once? I feel like this would have be an issue that appears before we get to local port elim.

Reply:
Nope, it doesnt

Reply:
So bundle elim does something other than simplify the signatures of components?

Reply:
Yep, it eliminates bundles throughout the entire ir at the moment, was I mistaken in how that was meant to work?

Reply:
Well, that blows up the encoding for local bundles unnecessarily. We need to eliminate signature bundles because we need to explicate the writes to input ports but local ports can just track writes to particular indices.

Reply:
Yeah it does as it's name says: it eliminates all bundles from the code.